### PR TITLE
Fix project canvas background color not respected

### PR DESCRIPTION
**Pull request by nirvn**
Title says it all; QField would always draw the canvas with a white background, irrespective of what was set by the opened project. This PR fixes this issue. 

Previously (broken) vs. PR:
image

### CODE REVIEW

**Comment by qfield-fairy** [comment]
Uploaded test apks for [armv7](https://download.opengis.ch/qfield/ci-builds/qfield-dev-643-3cc3b2f27149c7dfb26b7e5efe65ec0d42658cf3-armv7-X.apk) and [x86](https://download.opengis.ch/qfield/ci-builds/qfield-dev-643-3cc3b2f27149c7dfb26b7e5efe65ec0d42658cf3-x86-X.apk) (Android 5: [armv7](https://download.opengis.ch/qfield/ci-builds/qfield-dev-643-3cc3b2f27149c7dfb26b7e5efe65ec0d42658cf3-armv7-5.apk) and [x86](https://download.opengis.ch/qfield/ci-builds/qfield-dev-643-3cc3b2f27149c7dfb26b7e5efe65ec0d42658cf3-x86-5.apk))

File: src/qgsquick/qgsquickmapsettings.h [hunk 1]
```diff
@@ -163,6 +163,12 @@ class QgsQuickMapSettings : public QObject
     //! \copydoc QgsQuickMapSettings::rotation
     void setRotation( double rotation );
 
+    //! \copydoc QgsQuickMapSettings::backgroundColor
+    QColor backgroundColor() const;
```
Comment (m-kuhn) [inline src/qgsquick/qgsquickmapsettings.h]: If you add this as a `Q_PROPERTY` with `NOTIFY`, i will also be available in QML.

File: src/qgsquick/qgsquickmapsettings.h [hunk 2]
```diff
@@ -163,6 +163,12 @@ class QgsQuickMapSettings : public QObject
...
+
+    //! \copydoc QgsQuickMapSettings::setBackgroundColor
+    void setBackgroundColor( QColor &color );
```
Comment (m-kuhn) [inline src/qgsquick/qgsquickmapsettings.h]:
```suggestion
    void setBackgroundColor( const QColor &color );
```

File: src/qgsquick/qgsquickmapsettings.h (hunk 1 above)
Comment (nirvn) [reply src/qgsquick/qgsquickmapsettings.h]: Good, we could make use of that to set a background rectangle to match canvas bg color, that'll beautify panning et cie.

**Comment by nirvn** [comment]
@m-kuhn , Q_PROPERTY added. This unlocked the addition of a background color fill in QField, which makes panning around much nicer for non-white projects. Here's an ugly and to the point example:
Peek 2019-09-30 09-45

**Comment by nirvn** [comment]
@m-kuhn , on note: I couldn't get the background rectangle item's color to update dynamically even though the mapSettings.backgroundColor Q_PROPERTY does have a notification signal. Any clue as to what's happening here?

**Comment by qfield-fairy** [comment]
Uploaded test apks for [armv7](https://download.opengis.ch/qfield/ci-builds/qfield-dev-643-2a264b83a2a8c38b060e6113d1dde47273200c15-armv7-X.apk) and [x86](https://download.opengis.ch/qfield/ci-builds/qfield-dev-643-2a264b83a2a8c38b060e6113d1dde47273200c15-x86-X.apk) (Android 5: [armv7](https://download.opengis.ch/qfield/ci-builds/qfield-dev-643-2a264b83a2a8c38b060e6113d1dde47273200c15-armv7-5.apk) and [x86](https://download.opengis.ch/qfield/ci-builds/qfield-dev-643-2a264b83a2a8c38b060e6113d1dde47273200c15-x86-5.apk))

**Comment by nirvn** [comment]
A small additional fix has been appended to this PR whereas panning the map canvas would draw over the GPS information panel:
Peek 2019-09-30 11-16

With this fix, which clips content, things are looking good:
Peek 2019-09-30 11-17_good

**Comment by qfield-fairy** [comment]
Uploaded test apks for [armv7](https://download.opengis.ch/qfield/ci-builds/qfield-dev-643-b1450525657ea7af7d32b79f5e659e82802cf843-armv7-X.apk) and [x86](https://download.opengis.ch/qfield/ci-builds/qfield-dev-643-b1450525657ea7af7d32b79f5e659e82802cf843-x86-X.apk) (Android 5: [armv7](https://download.opengis.ch/qfield/ci-builds/qfield-dev-643-b1450525657ea7af7d32b79f5e659e82802cf843-armv7-5.apk) and [x86](https://download.opengis.ch/qfield/ci-builds/qfield-dev-643-b1450525657ea7af7d32b79f5e659e82802cf843-x86-5.apk))

File: src/qml/qgismobileapp.qml [hunk 3]
```diff
@@ -932,6 +939,7 @@ ApplicationWindow {
       onLoadProjectEnded: {
         busyMessage.visible = false
         openProjectDialog.folder = qgisProject.homePath
+        mapCanvasBackground.color = mapCanvas.mapSettings.backgroundColor
```
Comment (m-kuhn) [inline src/qml/qgismobileapp.qml]: Does it help if this line is removed?

```suggestion
        mapCanvasBackground.color = mapCanvas.mapSettings.backgroundColor
```

What this does is overwriting line 167 with a fixed value and "destroying" the dynamic binding.

Comment (nirvn) [reply src/qml/qgismobileapp.qml]: Sorry, I meant that with line 167 alone, the background color doesn't dynamically switch. Let me try again to make sure I'm not just plain wrong.

Comment (nirvn) [reply src/qml/qgismobileapp.qml]: Yep, just re-confirmed. The dynamic binding doesn't work, that line in onLoadProjectedEnded is needed.

Comment (m-kuhn) [reply src/qml/qgismobileapp.qml]: One more to add for the "let's get rid of all the warnings" project.
Ther's this hidden in the log

> qrc:/qml/qgismobileapp.qml:167: ReferenceError: mapSettings is not defined

But even now I can't get it to work dynamically, weird. Let's stick to the current solution if no one finds something.

**Comment by 3nids** [comment]
@nirvn when we merge this, can you backport upstream to QGIS Quick?

**Comment by nirvn** [comment]
@3nids , already pushing :)

**Comment by nirvn** [comment]
FYI, as far as I'm concerned, this is ready to merge.

**Comment by 3nids** [comment]
do we merge or would you like to investigate the property binding?

**Comment by nirvn** [comment]
I've tried some more, can't pinpoint the issue. @m-kuhn , do you have high hopes? :)
